mico-core/src/main/java/io/github/ust/mico/core/broker/MicoApplicationBroker.java

```diff
@@ -251,24 +251,30 @@ public MicoServiceDeploymentInfo addMicoServiceToMicoApplicationByShortNameAndVe
         } else {
             // Service already included, replace it with its newer version, ...
             MicoService existingMicoService = micoServices.get(0);
-            log.debug("MicoService '{}' is already included in application '{}' '{}' in version '{}'. " +
-                    "Replace it with the version '{}'.", serviceShortName, applicationShortName, applicationVersion,
-                existingMicoService.getVersion(), serviceVersion);
-
-            // Replace service in list of services in application
-            micoApplication.getServices().remove(existingMicoService);
-            micoApplication.getServices().add(micoService);
-
-            // Move the edge between the application and the service to the new version of the service
-            // by updating the corresponding deployment info
-            micoApplication.getServiceDeploymentInfos().stream()
-                .filter(sdi -> sdi.getService().getShortName().equals(serviceShortName))
-                .collect(Collectors.toList())
-                .forEach(sdi -> sdi.setService(micoService));
-
-            // Save the application with the updated list of services
-            // and service deployment infos in the database
-            applicationRepository.save(micoApplication);
+            if (!existingMicoService.getVersion().equals(serviceVersion)) {
```

Comment:
As a negation is usually less visible in java and the else part is shorter it would be more readable to switch it around.